packages/patternfly-4/react-core/src/components/Select/Select.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/FormControls/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** additional classes added to the Select control */
+  className: PropTypes.string,
+  /** value of selected option */
+  value: PropTypes.any,
+  /** Array of options OR array of option Groups available to select from.  If getOptionGroupLabel, it should be an array ofoption groups */
+  options: PropTypes.array,
+  /** flag indicating the options are grouped */
+  isOptionsGrouped: PropTypes.bool,
+  /** function to retrive the label for an option group */
+  getOptionGroupLabel: PropTypes.func,
+  /** function to retrive the array of options for an option group */
+  getOptionFromGroup: PropTypes.func,
+  /** function to retrive the label for an option */
+  getOptionLabel: PropTypes.func,
+  /** function to retrive the Value for an option */
+  getOptionValue: PropTypes.func,
+  /** function to retrive flag indicating if the option is disabled */
+  getOptionDisabled: PropTypes.func,
+  /** Flag indicating selection is valid */
+  isValid: PropTypes.bool,
+  /** Flag indicating the Select is disabled */
+  isDisabled: PropTypes.bool,
+  /** Optional callback for updating when selection loses focus */
+  onBlur: PropTypes.func,
+  /** Optional callback for updating when selection gets focus */
+  onFocus: PropTypes.func,
+  /** Optional callback for updating when selection changes */
+  onChange: PropTypes.func,
+  /** Custom flag to show that the Select requires an associated id or aria-label. */
+  'aria-label': props => {
+    if (!props.id && !props['aria-label']) {
+      return new Error('Select requires either an id or aria-label to be specified');
+    }
+    return null;
+  }
+};
+
+const defaultProps = {
+  className: '',
+  value: '',
+  options: null,
+  isOptionsGrouped: false,
+  getOptionGroupLabel: () => undefined,
+  getOptionFromGroup: () => undefined,
+  getOptionLabel: () => undefined,
+  getOptionValue: () => undefined,
+  getOptionDisabled: () => undefined,
+  isValid: true,
+  isDisabled: false,
+  onBlur: () => undefined,
+  onFocus: () => undefined,
+  onChange: () => undefined,
+  'aria-label': null
+};
+
+class Select extends React.Component {
+  /** call callback to handle selection change */
```

Comment:
More probably up to the group, but since these functions are well named and obvious comments are generally not needed. It is usually better to have function self-explanatory since comments add noise. I actually usually remove them from view in my editor when possible. Does not need to be changed now, but wanted to let you know you probably didn't need to provide these in the future.

Reply:
Sure.   I can remove them while I am making other updates.